sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -189,7 +203,11 @@ def split(
         stop_position=stop_position)
 
   def estimate_size(self):
-    file_names = [f for f in fileio.ChannelFactory.glob(self._pattern)]
+    if not self._pattern.is_accessible():
+      raise RuntimeError('value not accessible')
```

Comment:
`estimate_size` could be called at pipeline construction time. If that is the case and value is not accessible it should not raise an RuntimeError. (Could be the same for `_validate`). Please talk with @chamikaramj about what would be the best strategy in these cases.